File: structures-js/structures-api/src/api/idl/QueryOptionsC3Type.ts
import {IntC3Type, ObjectC3Type, StringC3Type} from '@kinotic/continuum-idl'

/**
 * Represents a {@link QueryOptions} type.
 * This is used to specify options that will be used when a Named Query is executed.
 */
export class QueryOptionsC3Type extends ObjectC3Type {

    public readonly type: string = 'queryOptions'

    constructor() {
        super('QueryOptions', 'org.kinotic.structures.domain')
        this.addProperty('timeZone', new StringC3Type())
        this.addProperty('requestTimeout', new IntC3Type())
        this.addProperty('pageTimeout', new StringC3Type())
    }
}

File: structures-core/src/main/java/org/kinotic/structures/api/domain/idl/QueryOptionsC3Type.java
package org.kinotic.structures.api.domain.idl;

import org.kinotic.continuum.idl.api.schema.FunctionDefinition;
import org.kinotic.continuum.idl.api.schema.IntC3Type;
import org.kinotic.continuum.idl.api.schema.ObjectC3Type;
import org.kinotic.continuum.idl.api.schema.StringC3Type;
import org.kinotic.structures.api.domain.QueryOptions;

/**
 * A {@link QueryOptionsC3Type} represents the {@link QueryOptions} type.
 * This allows us to detect its usage in {@link FunctionDefinition} parameters and apply the needed logic.
 * Created By Navíd Mitchell 🤪on 2/24/25
 */
public class QueryOptionsC3Type extends ObjectC3Type {

    public QueryOptionsC3Type() {
        addProperty("timeZone", new StringC3Type());
        addProperty("requestTimeout", new IntC3Type());
        addProperty("pageTimeout", new StringC3Type());
    }

}
